tree.go

```diff
@@ -930,13 +1037,54 @@ func (n *LeafNode) Hash() *Fr {
 
 func (n *LeafNode) Commitment() *Point {
 	if n.commitment == nil {
-		panic("nil commitment")
+		n.Commit()
 	}
 	return n.commitment
 }
 
-func (n *LeafNode) Commit() *Point {
-	return n.commitment
+var frPool = sync.Pool{
+	New: func() any {
+		ret := make([]Fr, NodeWidth)
+		return &ret
+	},
+}
+
+func (leaf *LeafNode) Commit() *Point {
+	if leaf.commitment == nil {
```

Comment:
Yeah let's not do that. Better implement the same thing as CoW that is done for internal nodes. Then it can use your method at flush time, and you don't have to put checks for `nil` commitments all over the code.

Reply:
Cool, let's wait for #295 and re-evaluate here.